exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/svremover/GenericCopierTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.physical.impl.svremover;
+
+import org.apache.drill.exec.memory.RootAllocator;
+import org.apache.drill.exec.record.BatchSchema;
+import org.apache.drill.test.rowSet.RowSet;
+import org.apache.drill.test.rowSet.RowSetBuilder;
+
+public class GenericCopierTest extends AbstractGenericCopierTest {
```

Comment:
One reason I was a bit hesitant to add this support earlier is that we need this to work for ALL Drill's data types, including the obscure ones. Clearly, this mechanism works for scalars. But, what about:

* Arrays of scalars
* Maps
* Arrays of maps
* Unions
* Lists of unions
* Lists of scalars
* Lists of maps
* Repeated lists of scalars
* Repeated lists of maps
* Repeated lists of repeated lists
* Repeated lists of unions

There are probably a few more that I'm overlooking. Until we test ALL of these, we can't be sure if the new code works as well as the old code.

Reply:
The generated code uses the copyFromSafe methods in code generated copiers. Your copyEntry method which is used by the GenericCopiers also uses the copyFromSafe method. So we should inherit all the same supported types. I think if all the unit tests, functional tests, and QA tests pass that will be a sufficient enough vote of confidence. Ideally we would have unit tests for all the operators (not just the copiers) which use the complex data types, but there is significant amount of work left to be able to effectively unit tests with complex data types and adding all the machinery to do so is out of the scope of this PR.